ddtrace/tracer/option.go

```diff
@@ -33,6 +35,65 @@ import (
+	"go.mongodb.org/mongo-driver/mongo":        {"MongoDB", "go.mongodb.org/mongo-driver", false},
+	"gocql/gocql":                              {"Cassandra", "github.com/gocql/gocql", false},
+	"gofiber/fiber.v2":                         {"Fiber", "github.com/gofiber/fiber", false},
+	"gomodule/redigo":                          {"Redigo", "github.com/gomodule/redigo", false},
+	"google.golang.org/api":                    {"Google API", "google.golang.org/api", false},
+	"google.golang.org/grpc":                   {"gRPC", "google.golang.org/grpc", false},
+	"google.golang.org/grpc.v12":               {"gRPC v12", "google.golang.org/grpc", false},
+	"gopkg.in/jinzhu/gorm.v1":                  {"Gorm (gopkg)", "github.com/jinzhu/gorm.v1", false},
+	"gorilla/mux":                              {"Gorilla Mux", "github.com/gorilla/mux", false},
+	"gorm.io/gorm.v1":                          {"Gorm v1", "gorm.io/gorm", false},
+	"graph-gophers/graphql-go":                 {"GraphQL", "github.com/graph-gophers/graphql-go", false},
+	"hashicorp/consul":                         {"Consul", "github.com/hashicorp/consul/api", false},
+	"hashicorp/vault":                          {"Vault", "github.com/hashicorp/vault/api", false},
+	"jinzhu/gorm":                              {"Gorm", "github.com/jinzhu/gorm", false},
+	"jmoiron/sqlx":                             {"SQLx", "github.com/jmoiron/sqlx", false},
+	"julienschmidt/httprouter":                 {"HTTP Router", "github.com/julienschmidt/httprouter", false},
+	"k8s.io/client-go/kubernetes":              {"Kubernetes", "k8s.io/client-go", false},
+	"labstack/echo":                            {"echo", "^github.com/labstack/echo$", false},
+	"labstack/echo.v4":                         {"echo v4", "github.com/labstack/echo/v4", false},
+	"miekg/dns":                                {"miekg/dns", "github.com/miekg/dns", false},
+	"net/http":                                 {"HTTP", "net/http", false},
+	"olivere/elastic":                          {"Elasticsearch", `gopkg\.in/olivere/elastic\.v(3|5)`, false},
+	"redis/go-redis.v9":                        {"Redis v9", "github.com/redis/go-redis/v9", false},
+	"segmentio/kafka.go.v0":                    {"Kafka v0", "github.com/segmentio/kafka-go", false},
+	"Shopify/sarama":                           {"Kafka (sarama)", "github.com/Shopify/sarama", false},
+	"sirupsen/logrus":                          {"Logrus", "github.com/sirupsen/logrus", false},
+	"syndtr/goleveldb/leveldb":                 {"LevelDB", "github.com/syndtr/goleveldb", false},
+	"tidwall/buntdb":                           {"BuntDB", "github.com/tidwall/buntdb", false},
+	"twitchtv/twirp":                           {"Twirp", "github.com/twitchtv/twirp", false},
+	"urfave/negroni":                           {"Negroni", "github.com/urfave/negroni", false},
+	"zenazn/goji.v1/web":                       {"Goji", "github.com/zenazn/goji", false},
```

Comment:
Will this have to be updated every time a new integration is added? If so, I'm wondering if there's a way we can make a script + test to do this automatically (or at least make sure we don't forget it when we introduce a new contrib library)

Reply:
Yeah, unfortunately it would have to be updated each time :/ I considered a walk through the `contrib` package to get each of the integrations, but `gName` isn't very easy to parse directly from the contrib name. Though I suppose it could be done with several if statements.

I.e. if statement to cut `github.com/` and `gopkg.in/`, to replace `/v` with `.v`, etc... 

Thoughts?

Reply:
Ah, something else I considered that I forgot to mention. The contrib package seems to have a [naming schema](https://pkg.go.dev/gopkg.in/DataDog/dd-trace-go.v1/contrib#section-readme:~:text=The%20naming%20convention%20for%20the%20integration%20packages%20is%3A) that should allow us to easily go from the standard library name to the contrib package name, but I wasn't entirely sure if this was being followed 100%.

For example, it says "All new integrations should be suffixed with .vN where N is the major version that is being covered. If the integration covers more than one major version, the minimum version supported should be chosen for the suffix. " So for integrations like `olivere/elastic`, which supports v3 and v5, I expected the component name to be `olivere/elastic.v3`, but it's just,,, `olivere/elastic`. 

Maybe I'm misinterpreting, but I was pretty wary of mistakes in namings that I just... made a map lol. Sorry for the long essay!

Reply:
Yeah I think the naming schema arrived after a few packages already had names that didn't match the schema (so unfortunately there's a few where the name doesn't align). I do think we should try to have a test to catch cases where developers forget to add this line to their new contrib library as it's a very easy mistake to make and a pretty difficult one to catch. We don't necessarily have to verify the exact text is as expected, but perhaps something like `TestTelemetryEnabled` which verifies that every contrib package at least _imports_ the telemetry package so it's likely the developer remembered. In our case here we can't do that exact thing since every package imports tracer, but perhaps a similar approach here that walks over each folder looking for at least one call to `ImportIntegration` ?